src/libAtomVM/externalterm.c

```diff
@@ -536,6 +600,20 @@ static int calculate_heap_usage(const uint8_t *external_term_buf, size_t remaini
             return term_boxed_integer_size(value);
         }
 
+        case SMALL_BIG_EXT: {
+            uint8_t num_bytes = external_term_buf[1];
+            if (UNLIKELY(num_bytes > 8 || remaining < (SMALL_BIG_EXT_BASE_SIZE + num_bytes))) {
+                return INVALID_TERM_SIZE;
+            }
+            uint8_t sign = external_term_buf[2];
+            *eterm_size = SMALL_BIG_EXT_BASE_SIZE + num_bytes;
+            avm_uint64_t value = read_bytes(external_term_buf + 3, num_bytes);
+            if (UNLIKELY((sign == 0 && value > INT64_MAX) || (sign != 0 && value > (((avm_uint64_t) INT64_MAX) + 1)))) {
```

Comment:
Just add a comment here above to mention that read_bytes returns an avm_uint64_t, and negatives are in 2-complement.
The code is correct but not super easy to understand.